setup.py

```diff
@@ -0,0 +1,56 @@
+"""
+automated installation instructions
+moving away from requirements.txt
+"""
+from setuptools import find_packages, setup
+
+setup(
+    name='automated-interpretation-pipeline',
+    version='0.0.1',
+    author='Matthew Welland, CPG',
+    author_email=(
+        'matthew.welland@populationgenomics.org.au, '
+        'cas.simons@populationgenomics.org.au'
+    ),
+    url='https://github.com/populationgenomics/automated-interpretation-pipeline',
+    description='Variant annotation and prioritisation process',
+    license='MIT',
+    classifiers=[
+        'Environment :: Console',
+        'Intended Audience :: Science/Research',
+        'License :: OSI Approved :: MIT License',
+        'Natural Language :: English',
+        'Operating System :: MacOS :: MacOS X',
+        'Operating System :: POSIX',
+        'Operating System :: Unix',
+        'Programming Language :: Python',
+        'Topic :: Scientific/Engineering',
+        'Topic :: Scientific/Engineering :: Bio-Informatics',
+    ],
+    packages=find_packages(),
+    include_package_data=True,
+    install_requires=[
+        'click==8.0.4',
+        'cloudpathlib[azure,gs]==0.7.0',
+        'cpg-utils==4.4.0',
+        'cyvcf2==0.30.14',
+        'dill==0.3.4',
+        'hail==0.2.93',
+        'Jinja2==3.0.3',
+        'pandas==1.3.5',
+        'peddy==0.4.8',
+        'requests==2.25.1',
+        'sample-metadata==4.10.0',
+        'seqr-loader==1.2.5',
+    ],
+    extras_require={
```

Comment:
I like it! Will adopt `extras_require` this in my projects as well :D

`pre-commit` section seems to duplicate `requirements-lint.txt`, maybe worth reading `requirements-lint.txt` contents here, to avoid them accidentally getting out of sync?

Reply:
@vladsaveliev that was exactly my thinking! Then I started reading around for the best way to parse requirements into setup.py and the suggestions were to keep them separate (e.g. https://stackoverflow.com/a/16624700)

My main reason for keeping requirements-lint.txt separate is that to run linting in CI it only needs those 3 packages, but running `pip install .[lint]` would install numpy, hail, etc...

Maybe I should remove the `extras` option for lint...

Reply:
This was a longer explanation https://stackoverflow.com/a/33685899 - I'd never used them in this separate way before

Reply:
> y main reason for keeping requirements-lint.txt separate is that to run linting in CI it only needs those 3 packages, but running pip install .[lint] would install numpy, hail, etc...

That's right, `requirements-lint.txt` is certainly needed!

Reply:
> Maybe I should remove the extras option for pre-commit...

I think that would make sense! 

But it's really nitpicking from me, feel free to merge as is :-)